Mathlib/Algebra/BigOperators/Finsupp.lean

```diff
@@ -534,11 +534,23 @@ lemma sum_cons [AddCommMonoid M] (n : ℕ) (σ : Fin n →₀ M) (i : M) :
 
 lemma sum_cons' [AddCommMonoid M] [AddCommMonoid N] (n : ℕ) (σ : Fin n →₀ M) (i : M)
     (f : Fin (n+1) → M → N) (h : ∀ x, f x 0 = 0) :
-    (sum (Finsupp.cons i σ) f) = f 0 i + sum σ (Fin.tail f) := by
+    (sum (cons i σ) f) = f 0 i + sum σ (Fin.tail f) := by
   rw [sum_fintype _ _ (fun _ => by apply h), sum_fintype _ _ (fun _ => by apply h)]
   simp_rw [Fin.sum_univ_succ, cons_zero, cons_succ]
   congr
 
+lemma sum_insertNth [AddCommMonoid M] {n : ℕ} (σ : Fin n →₀ M) (i : M) (p : Fin (n + 1)) :
+    (insertNth p i σ).sum (fun _ e ↦ e) = i + σ.sum (fun _ e ↦ e) := by
+  rw [sum_fintype _ _ (fun _ => rfl), sum_fintype _ _ (fun _ => rfl)]
+  exact Fin.sum_insertNth i σ p
+
+lemma sum_insertNth' [AddCommMonoid M] [AddCommMonoid N] {n : ℕ} (σ : Fin n →₀ M) (i : M)
+    (p : Fin (n + 1)) (f : Fin (n+1) → M → N) (h : ∀ x, f x 0 = 0) :
+    sum (insertNth p i σ) f = f p i + sum σ (Fin.removeNth p f) := by
```

Comment:
Isn't the first one of these just the second in the special case that `f := fun _ ↦ id`? If so, I would rename the first one as `sum_insertNth_id`, put it after the other one, and rename the other one to `sum_insertNth`.